### ISSUE TITLE
Session->set_cookie(): Use of PHP's setcookie()

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#2659]
In this function, there is:
```
return setcookie($name, $value, 'expires='.$expire.';path='.$params['path'].';domain='.$params['domain'].';secure='.$params['secure'].';httponly='.$params['httponly'].';'.$same_site);
```
According to the function signature, each placement is defined:
```
1st: name of cookie (string)
2nd: value of cookie name (string)
3rd: expiration of cookie (int)
4th: path of cookie (string)
5th: domain of cookie (string)
6th: is secure? (bool)
7th: is httponly? (bool)
```
What is given above is that:
```
3rd: a long string (not an integer)
```
I believe this will cause the `setcookie()` function to fail.

**Comment by bhsmither** [comment]
See: https://stackoverflow.com/questions/39750906/php-setcookie-samesite-strict
for some interesting hacks that deal with effective handling of PHP <=7.2 versus PHP >=7.3

**Comment by abrookbanks** [comment]
It's using the options method to have more control.

![Screenshot 2020-11-13 095827](https://user-images.githubusercontent.com/4520803/99059958-36866980-2597-11eb-93f7-480575c3d503.png)

**Comment by bhsmither** [comment]
Please review the statement. The third parameter is a string, not an array.

Additionally, this particular function signature for setcookie(string, string, array) is available **only** for PHP 7.3 and higher. Please keep in mind CubeCart is still claiming it will work on PHP 5.5+ (5.4+ as the 'title' in setup).

**Comment by abrookbanks** [comment]
Resolved within #2689
